composer.json

```diff
@@ -20,15 +20,24 @@
             "homepage": "https://github.com/friendsofsymfony/FOSHttpCache/contributors"
         }
     ],
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "https://github.com/ddeboer/guzzle6-adapter"
+        }
+    ],
     "require": {
-        "php": ">=5.3.3",
-        "guzzle/guzzle": "~3.8",
+        "php": ">=5.4.8",
         "symfony/event-dispatcher": "~2.3",
-        "symfony/options-resolver": "~2.3"
+        "symfony/options-resolver": "~2.3",
+        "php-http/adapter": "dev-internal_separation",
+        "guzzlehttp/psr7": "~1.0"
```

Comment:
It is tricky. ;)

We have a message factory interface which allows you to pass a custom implementation. However, you don't have to. We have a MessageFactoryGuesser implementation which autodiscovers a few common implementations (currently guzzle and zend).

Also, the message factory is only needed for non PSR-7 adapters.